lib/parse.js

```diff
@@ -197,6 +207,21 @@ define(['esprima', './lang'], function (esprima, lang) {
         return result || null;
     }
 
+    /**
+     * Just parses the text to an AST to be used in other methods.
+     * @param {String} fileName
+     * @param {String} fileContents
+     * @param {Object} options
+     * @returns {Object} AST
+     */
+    parse.parseFileContents = function (fileName, fileContents, options) {
+        var localOptions = options || {};
+        return esprima.parse(fileContents, {
+            range: localOptions.range,
+            loc: localOptions.loc
+        });
+    }
```

Comment:
I like keeping the changes you have for the `parse` methods to take an AST Object or a JS String, and just skips the esprima call if the passed in value is the AST Object. However I wonder if we need this particular method in the API.

I would prefer to not have an API that is basically "call esprima". If it helps, maybe we can expose the version of esprima used by this package as `parse.esprima` if you wanted to make sure to use a version of esprima that matches the AST structure that `parse` works on.

Reply:
Yes, I admit, that saving the `esprima` reference in `vscode-requirejs` and making `amodro-trace` be responsible for parsing too was a little selfish ;-) On the other hand, if `amodro-trace` provides AST traversing and analysis, why not parsing too?

I don't insist on having the `parse` method here. I'm it and the related code. I'll put it to `vscode-requirejs` and use `esprima` too.